cmd/skywire-visor/commands/root.go

```diff
@@ -262,6 +268,33 @@ func runVisor(conf *visorconfig.V1) {
 			conf.Hypervisors = append(conf.Hypervisors, pubkey)
 		}
 	}
+	//autopeering should only happen when there is no local or remote hypervisor set in the config.
+	if conf.Hypervisors != nil {
+		if conf.Hypervisor != nil {
+			autopeer = false
+		}
+	}
+	if autopeer {
+		log.Infof("autopeering...")
+		var hvkey string
+		hvkey, err := script.Exec(peercmd).String() //TODO: mrpalide this needs to run again on loss of connection to the hypervisor within the same constraints
+		if err != nil {
+			log.Error("error autopeering")
+		} else {
+			hvkey = strings.TrimSuffix(hvkey, "\n")
+			hypervisorPKsSlice := strings.Split(hvkey, ",")
+			for _, pubkeyString := range hypervisorPKsSlice {
+				if err := pubkey.Set(pubkeyString); err != nil {
+					log.Warnf("Cannot add %s PK as remote hypervisor PK due to: %s", pubkeyString, err)
+					continue
+				}
+				log.Infof("%s PK added as remote hypervisor PK", pubkeyString)
+				conf.Hypervisors = append(conf.Hypervisors, pubkey)
+				skyenv.AutoPeer = true
```

Comment:
What is this doing?

Reply:
adding the public key to the remote hypervisors in the visor conf